hydrolib/core/io/net/models.py

```diff
@@ -0,0 +1,916 @@
+from __future__ import annotations
+
+from datetime import datetime
+from pathlib import Path
+from platform import uname
+from typing import List, Optional, Union, Dict
+from _pytest.mark import structures
+
+import meshkernel as mk
+from meshkernel.py_structures import GeometryList
+import netCDF4 as nc
+import numpy as np
+from numpy.lib import poly
+from pydantic import Field
+from pydantic.types import PyObject
+from pydantic.utils import ValueItems
+
+from hydrolib.core import __version__
+from hydrolib.core.basemodel import BaseModel
+from hydrolib.core.io.net.reader import UgridReader
+from hydrolib.core.io.net.writer import UgridWriter
+
+import matplotlib
+from matplotlib.collections import LineCollection
+
+
+def split_by(gl: mk.GeometryList, by: float) -> list:
+    """Function to split mk.GeometryList by seperator."""
+    splitidx = np.full(len(gl.x_coordinates), False)
+    x, y = gl.x_coordinates.copy(), gl.y_coordinates.copy()
+    idx = np.where(x == by)[0]
+
+    xparts = np.split(x, idx)
+    yparts = np.split(y, idx)
+
+    lists = [
+        mk.GeometryList(xp[min(i, 1) :], yp[min(i, 1) :])
+        for i, (xp, yp) in enumerate(zip(xparts, yparts))
+    ]
+
+    return lists
+
+
+class Mesh2d(BaseModel):
+
+    meshkernel: Optional[mk.MeshKernel] = Field(default_factory=mk.MeshKernel)
+
+    mesh2d_node_x: np.ndarray = np.empty(0, dtype=np.double)
+    mesh2d_node_y: np.ndarray = np.empty(0, dtype=np.double)
+    mesh2d_node_z: np.ndarray = np.empty(0, dtype=np.double)
+
+    mesh2d_edge_x: np.ndarray = np.empty(0, dtype=np.double)
+    mesh2d_edge_y: np.ndarray = np.empty(0, dtype=np.double)
+    mesh2d_edge_z: np.ndarray = np.empty(0, dtype=np.double)
+    mesh2d_edge_nodes: np.ndarray = np.empty((0, 2), dtype=np.int32)
+
+    mesh2d_face_x: np.ndarray = np.empty(0, dtype=np.double)
+    mesh2d_face_y: np.ndarray = np.empty(0, dtype=np.double)
+    mesh2d_face_z: np.ndarray = np.empty(0, dtype=np.double)
+    mesh2d_face_nodes: np.ndarray = np.empty((0, 0), dtype=np.int32)
+
+    def empty(self):
+        return self.mesh2d_node_x.size == 0
```

Comment:
```suggestion
    def empty(self) -> bool:
        return self.mesh2d_node_x.size == 0
```

Nitpicking: I would personally always prefix boolean functions with either `is_` or `has_`, in this case `is_empty`.